packages/runtime/container-runtime/src/summary/documentSchema.ts

```diff
@@ -68,6 +69,15 @@ export interface IDocumentSchema {
 	refSeq: number;
 
 	runtime: Record<string, DocumentSchemaValueType>;
+
+	/**
+	 * minVersionForCollab that was used when this document was created.
```

Comment:
You keep mentioning about document creation, maybe I'm missing something. I thought this minVersion would periodically creep up over time as newer and newer code with higher and higher min versions opens the doc.  In parallel with the specific config settings getting locked in as they light up.

Reply:
Ok I see what's going on. Based on how you've added this here, you don't need to be involving DocumentSchemaController at all.  Just put this on the metadata blob - see `ContainerRuntime.addMetadataToSummary`.

That will lock the value down to whatever was in effect when the document was first summarized. This is a fine lower bound for `minVersionForCollab`, and seems like what you're targeting for this PR.

Let's discuss the approach I mentioned above as a potential follow-up?

Reply:
> I thought this minVersion would periodically creep up over time as newer and newer code with higher and higher min versions opens the doc

Yes, this is how it should work. I think it may make sense to "keep" the largest `minVersionForCollab` we see. For example, if we start out with `minVersionForCollab = "2.20.0"`, we shouldn't update if a new client joins with `minVersionForCollab = "2.20.0"`. But if a new client joins with `minVersionForCollab = "2.40.0"`, then we should update it going forward. I think this makes the most sense with real world scenarios

> Based on how you've added this here, you don't need to be involving DocumentSchemaController at all.

Good point, I'll try this approach

Reply:
Cool. Let's discuss offline, I have some more thoughts.